### ISSUE TITLE
精度验证的两组实验

### ISSUE THREAD

**Issue by ccssu** [Oneflow-Inc/one-yolov5#38]
两组实验 

### 第2组 验证sync-bn对实验精度的影响。pytorch版本yolov5开sync-bn，再训练一次。
在前面https://github.com/Oneflow-Inc/one-yolov5/issues/37#issue-1384553187   ,    https://github.com/Oneflow-Inc/one-yolov5/issues/26#issuecomment-1251671595 两次实验结果中oneflow 和 pytorch 版本yolov5的map值是对齐的，由于oneflow训练的结果开了sync-bn 而 pytorch版本没有开，现在添组对于sync-bn是否影响训练结果的验证实验。

实时数据在下面更新。

**Comment by ccssu** [comment]
# 第一组实验 

## 环境

机器
- oneflow25-root

配置项: *详细数据可见日志 [run/train/exp4/opt.yaml]( https://paste.ubuntu.com/p/qVT9wkvzYb/ )*

-  shuffle =  True
-   sync-bn = False
-   amp = True 
-  --workers 8

## 数据 
目前opochs:  147
 | |metrics/mAP_0.5|metrics/mAP_0.5:0.95|
 | --- | --- | --- |
 |当前训练值| 38.109 | 22.277 |
| 目标值 |  45.6 | 27.7 | 
|当前训练值 - 目标值|-7.491 | -5.423|

**Comment by ccssu** [comment]
# 第二组实验 

## 环境

机器
- oneflow27-root

配置项: *详细数据可见日志 [run/train/exp4/opt.yaml](https://paste.ubuntu.com/p/YbcPDgNW6h/ )*

-   shuffle =  True
-   sync-bn = False
-   amp = False # 使用Fp32训练
-  --workers 8

## 数据 

刚刚没有放在tmux中运行，以刚刚会议中同样的指令在tmux启动。启动参数可以[run/train/exp4/opt.yaml](https://paste.ubuntu.com/p/YbcPDgNW6h/ )*

**Comment by ccssu** [comment]
|epochs数|框架| 几张卡|shuffle | sycn-bn | amp |workers |metrics/mAP_0.5|metrics/mAP_0.5:0.95|
|---| ---| ---| --- | --- | --- | --- | --- | --- |
| 125 | torch | 单卡 | true | false| true | 8 |  38.956 |             22.966|
|125 | torch | 两卡 | true | false | true| 8 |   37.392 |             21.928 | 
|125 | torch | 两卡 | true | false | false | 8 |   37.479 |            21.811 | 
| 125 | torch | 两卡 | true | true | false | 8 |  37.512 |             21.898 |   



就目前的数据，猜测 $mAP$值 受到使用多卡的影响。

**Comment by BBuf** [comment]
> epochs数 	框架 	几张卡 	shuffle 	sycn-bn 	amp 	workers 	metrics/mAP_0.5 	metrics/mAP_0.5:0.95
> 125 	torch 	单卡 	true 	false 	true 	8 	38.956 	22.966
> 125 	torch 	两卡 	true 	false 	true 	8 	37.392 	21.928
> 125 	torch 	两卡 	true 	false 	false 	8 	37.479 	21.811
> 125 	torch 	两卡 	true 	true 	false 	8 	37.512 	21.898
> 
> 以上的数据都是使用 pytorch 版本yolov5的 48a85314bc80d8023c99bfb114cea98d71dd0591（_commit版本_) 跑出的结果。 就目前的数据，猜测 mAP值 受到使用多卡的影响。

没事，我们对齐一下pytorch用8卡训练的300epoch的精度就可以了，上述实验已经基本证明无论是amp还是sync-bn都对精度几乎没有影响，我们可以放心的做最后一组实验了。即实验oneflow ddp8卡的精度是否能达到pytorch ddp8卡的精度，如果可以我们的实验就做到这里吧。

**Comment by Ldpe2G** [comment]
> 没事，我们对齐一下pytorch用8卡训练的300epoch的精度就可以了，上述实验已经基本证明无论是amp还是sync-bn都对精度几乎没有影响，我们可以放心的做最后一组实验了。即实验oneflow ddp8卡的精度是否能达到pytorch ddp8卡的精度，如果可以我们的实验就做到这里吧。

现在每张卡 batch size 是多少，和官方设置是一样的吗

如果官方放出来的map是用8卡跑的，那还不能说明 sync-bn 无效，因为卡用的越多，用sync-bn，相当于计算mean 和 var的样本就越多，可能会对精度有提升，之前的实验只用了两卡。

**Comment by BBuf** [comment]
> > 没事，我们对齐一下pytorch用8卡训练的300epoch的精度就可以了，上述实验已经基本证明无论是amp还是sync-bn都对精度几乎没有影响，我们可以放心的做最后一组实验了。即实验oneflow ddp8卡的精度是否能达到pytorch ddp8卡的精度，如果可以我们的实验就做到这里吧。
> 
> 现在每张卡 batch size 是多少，和官方设置是一样的吗
> 
> 如果官方放出来的map是用8卡跑的，那还不能说明 sync-bn 无效，因为卡用的越多，用sync-bn，相当于计算mean 和 var的样本就越多，可能会对精度有提升，之前的实验只用了两卡。

我们的硬件环境是不支持batch_size完全对齐的，官方的checkpoint的bacth_size是64，我们的显卡只能支持到16，我们接下来做一组对齐batch_size的8卡实验验证就可以了。都使用fp32 && sync-bn && bs=16 的完全一致的超参数配置。

**Comment by Ldpe2G** [comment]
> 我们的硬件环境是不支持batch_size完全对齐的，官方的checkpoint的bacth_size是64，我们的显卡只能支持到16，我们接下来做一组对齐batch_size的8卡实验验证就可以了。都使用fp32 && sync-bn && bs=16 的完全一致的超参数配置。

我们租了一台A100，可以到上面去训，显存足够大 40G，可以找ouyang开下权限

**Comment by BBuf** [comment]
> > 我们的硬件环境是不支持batch_size完全对齐的，官方的checkpoint的bacth_size是64，我们的显卡只能支持到16，我们接下来做一组对齐batch_size的8卡实验验证就可以了。都使用fp32 && sync-bn && bs=16 的完全一致的超参数配置。
> 
> 我们租了一台A100，可以到上面去训，显存足够大 40G，可以找ouyang开下权限

好的，那可以。
